tests/testthat/test-coh_match_utils.R

```diff
@@ -0,0 +1,244 @@
+  data_set = sample_cohort_s,
+  col_names = "vaccine_date_2",
+  status = c("v", "u")
+)
+
+# Test for basic expectations - all provided
+test_that("`match_cohort`: basic expectations - all provided", {
+  matched_cohort <- match_cohort_(
+    data_set = sample_cohort_s,
+    vacc_status_col = "vaccine_status",
+    exact = "sex",
+    nearest = c(age = 1)
+  )
+  # expect dataframe
+  expect_s3_class(matched_cohort, "data.frame")
+
+  # keep all the columns and add only "subclass"
+  expect_true(
+    all(
+      names(matched_cohort) %in%
+        c(names(sample_cohort_s), "subclass")
+    )
+  )
+
+  # even number of pairs
+  expect_setequal(
+    nrow(matched_cohort),
+    length(unique(matched_cohort$subclass)) * 2
+  )
+
+  # same number of vaccinated and unvaccinated
+  expect_identical(
+    nrow(matched_cohort[matched_cohort$vaccine_status == "u", ]),
+    nrow(matched_cohort[matched_cohort$vaccine_status == "v", ])
+  )
+
+  # same number of sex in "v" and "u"
+  expect_identical(
+    table(matched_cohort[matched_cohort$vaccine_status == "u", ]$sex),
+    table(matched_cohort[matched_cohort$vaccine_status == "v", ]$sex)
+  )
+
+  # same number of sex in "v" and "u"
+  expect_identical(
+    table(matched_cohort[matched_cohort$vaccine_status == "u", ]$sex),
+    table(matched_cohort[matched_cohort$vaccine_status == "v", ]$sex)
+  )
+
+  # ages are not equal
+  expect_false(
+    isTRUE(
+      all.equal(
+        table(matched_cohort[matched_cohort$vaccine_status == "u", ]$age),
+        table(matched_cohort[matched_cohort$vaccine_status == "v", ]$age)
+      )
+    )
+  )
+})
+
+# Test for exact match
+test_that("`match_cohort`: exact match", {
+  matched_cohort <- match_cohort_(
+    data_set = sample_cohort_s,
+    vacc_status_col = "vaccine_status",
+    exact = "sex"
+  )
+  # same number of categories in "v" and "u"
+  expect_identical(
+    table(matched_cohort[matched_cohort$vaccine_status == "u", ]$sex),
+    table(matched_cohort[matched_cohort$vaccine_status == "v", ]$sex)
+  )
+})
+
+# Test for nearest match
+test_that("`match_cohort`: nearest match", {
+  matched_cohort <- match_cohort_(
+    data_set = sample_cohort_s,
+    vacc_status_col = "vaccine_status",
+    nearest = c(age = 3)
+  )
+  # even number of pairs
+  expect_setequal(
+    nrow(matched_cohort),
+    length(unique(matched_cohort$subclass)) * 2
+  )
+
+  # same number of vaccinated and unvaccinated
+  expect_identical(
+    nrow(matched_cohort[matched_cohort$vaccine_status == "u", ]),
+    nrow(matched_cohort[matched_cohort$vaccine_status == "v", ])
+  )
+
+  # ages are not equal due to caliper
+  expect_false(
+    isTRUE(
+      all.equal(
+        table(matched_cohort[matched_cohort$vaccine_status == "u", ]$age),
+        table(matched_cohort[matched_cohort$vaccine_status == "v", ]$age)
+      )
+    )
+  )
+})
+
+#### Tests for match_pair_info() ####
+# Assign vaccination status
+sample_cohort_l$vaccine_status <- set_status(
```

Comment:
Again, I recommend moving all variables to the beginning of each file. That way also reduces the probability of creating duplicated data, since all of it is concentrated in the same place in the file.

Reply:
take a look at all of the other test files to fix this.

Reply:
Solved in 4f50a0fff806fcbb500ae61cba95e60d83276a4d